account_due_list/account_move_line.py

```diff
@@ -0,0 +1,166 @@
+#    Copyright (c) 2008 Zikzakmedia S.L. (http://zikzakmedia.com)
+#                       Jordi Esteve <jesteve@zikzakmedia.com>
+#
+#    Copyright (C) 2011 Domsense srl (<http://www.domsense.com>)
+#    Copyright (C) 2011-2013 Agile Business Group sagl
+#    (<http://www.agilebg.com>)
+#    Ported to Odoo by Andrea Cometa <info@andreacometa.it>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import fields, orm
+from openerp.tools.translate import _
+
+
+class account_move_line(orm.Model):
+
+    def _get_invoice(self, cr, uid, ids, field_name, arg, context=None):
+        invoice_pool = self.pool['account.invoice']
+        res = {}
+        for line in self.browse(cr, uid, ids, context=context):
+            inv_ids = invoice_pool.search(
+                cr,
+                uid,
+                [('move_id', '=', line.move_id.id)],
+                context=context
+            )
+            if len(inv_ids) > 1:
+                raise orm.except_orm(
+                    _('Error'),
+                    _('Inconsistent data: move %s has more than one invoice')
+                    % line.move_id.name
+                )
+            if inv_ids:
+                res[line.id] = inv_ids[0]
+            else:
+                res[line.id] = False
+        return res
+
+    def _get_day(self, cr, uid, ids, field_name, arg, context=None):
+        res = {}
+        for line in self.browse(cr, uid, ids, context=context):
+            if line.date_maturity:
+                res[line.id] = line.date_maturity
+            else:
+                res[line.id] = False
+        return res
+
+    def _get_move_lines(self, cr, uid, ids, context=None):
+        invoice_pool = self.pool['account.invoice']
+        res = []
+        for invoice in invoice_pool.browse(cr, uid, ids, context=context):
+            if invoice.move_id:
+                for line in invoice.move_id.line_id:
+                    if line.id not in res:
+                        res.append(line.id)
+        return res
+
+    _inherit = 'account.move.line'
+
+    _columns = {
+        'invoice_origin': fields.related(
+            'invoice',
+            'origin',
+            type='char',
+            string='Source Doc',
+            store=False
```

Comment:
This is redundant, as store=False is the default. You don't need also to put one argument per line. It takes a lot of lines and doesn't improve readibility.